oxidation/crates/platform-async/src/task.rs

```diff
@@ -0,0 +1,17 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
+
+use std::future::Future;
+
+pub trait Taskable<T>: Future<Output = T> {
```

Comment:
What's the need for this trait ?

I guess this is to abstract the fact we have 2 implementations of Task (desktop vs web). If this is the fact I think we are better without any trait at all.
This is the approach used for the crate `parsec_api_crypto` (that have rustcrypto and libsodium implementations).

This has be benefit of simplifying the code (no need to use the trait in the caller code, simpler types to work with, async traits add complexity)
The only drawback is it makes impossible to use both implementations at the same time, but this is not a concern in here ^^

Reply:
It's to make a sure that any implementations will share a set of method